frontend/src/employee-frontend/components/UnitPage.tsx

```diff
@@ -49,6 +55,51 @@ const UnitPage = React.memo(function UnitPage({ id }: { id: UUID }) {
 
   const [openGroups, setOpenGroups] = useState<Record<string, boolean>>({})
 
+  const history = useHistory()
+  const queryParams = useCallback(
+    () => new URLSearchParams(location.search),
+    []
+  )
+
+  const getOpenGroupsFromQueryParams = useCallback((): Record<
+    string,
+    boolean
+  > => {
+    const qp = queryParams().get('open_groups')
+    if (qp != null) {
+      return qp
+        .split(',')
+        .reduce(
+          (prev, cur) => (cur ? Object.assign(prev, { [cur]: true }) : prev),
+          {}
+        )
+    } else {
+      return {}
+    }
+  }, [queryParams])
+
+  useEffect(() => {
+    setOpenGroups(getOpenGroupsFromQueryParams())
+  }, [getOpenGroupsFromQueryParams])
+
+  const openGroupsToStringList = (
+    openGroups: Record<string, boolean>
+  ): string => {
+    return Object.keys(openGroups)
+      .reduce(
+        (prev, cur) => (openGroups[cur] == true ? prev.concat(cur) : prev),
+        []
+      )
+      .join(',')
+  }
+
+  useEffect(() => {
+    const openList = openGroupsToStringList(openGroups)
+    openList.length > 0
+      ? history.push(`?open_groups=${openList}`)
+      : history.push('?')
+  }, [openGroups, history])
+
```

Comment:
The state is now stored in two places: URL and React component state.

Could it be like this instead (partly pseudocode):
```
const openGroups = useMemo(
  () => getOpenGroupsFromQueryParams(location.search),
  [location.search]
)
const setOpenGroups = useCallback((openGroups: Record<string, boolean>) => 
  history.push(`?open_groups=${openGroupsToStringList(openGroups)}`)
)
```

The functions `getOpenGroupsFromQueryParams` and `openGroupsToStringList` could be moved outside of the component function, too.

Reply:
I give a +1 to that idea. We have a `useQuery` hook that is useful in this case. 

Also, I think `history.replace` would be a better idea than `history.push` so that browser history doesn't get filled up from simply opening and closing group collapsibles.

Reply:
Without the component state the open groups are lost when navigating between the unit tabs. I added a test for this case.